thunder/benchmarks/benchmark_litgpt.py

```diff
@@ -74,7 +77,46 @@ def __init__(
         self.shard_mode = shard_mode
         self.bucketing_mode = bucketing_mode
         self.sharding_size = sharding_size
+        self.ddp_bucket_size = ddp_bucket_size
+        self.fsdp_bucket_params = fsdp_bucket_params
         self.micro_batch_size = micro_batch_size
+
+        # Clarify benchmark assumptions
+        if self.sharding_size is not None:
+            assert (
+                "thunder" not in self.compile
+            ), "Hybrid Sharding (FSDP/DP) using --sharding_size is not yet supported for Thunder. Coming soon."
+
+            assert self.shard_mode in [
+                "hybrid_zero2",
+                "hybrid_zero3",
+            ], "Sharding Size is only used with Hybrid FSDP/DP style parallelism. Please "
+
+            assert (
+                world_size % self.sharding_size == 0
+            ), f"World size {world_size} is not divisible by Hybrid Sharding Size {self.sharding_size}"
+
+        if self.bucketing_mode != "none" and self.distributed_mode != "fsdp":
+            print(
+                f"[WARNING] --bucketing_mode {self.bucketing_mode} will be ignored as \
+             it is only used for FSDP style parallelism but running {self.distributed_mode}"
+            )
+
+        assert not (
+            "thunder" in self.compile and self.bucketing_mode == "size"
+        ), "'size' bucketing mode is not supported for Thunder. Please use 'none' or 'block'."
+
+        if self.fsdp_bucket_params is not None:
+            if self.distributed_mode != "fsdp":
+                print(
+                    f"[WARNING] Found --fsdp_bucket_params but Distributed mode is {self.distributed_mode}. Will be ignnored"
+                )
+
+            if self.bucketing_mode != "size":
+                print(
+                    f"[WARNING] Bucketing mode is set to {self.bucketing_mode}. --fsdp_bucket_params will be ignoted."
```

Comment:
```suggestion
                    f"[WARNING] Bucketing mode is set to {self.bucketing_mode}. --fsdp_bucket_params will be ignored."
```

Reply:
I can't seem to spell ignored haha! Thanks for catching these.